tests/unittests/tests-ecc/tests-ecc.c

```diff
@@ -75,11 +76,36 @@ static void test_hamming256_padding(void)
     TEST_ASSERT_EQUAL_INT(Hamming_ERROR_ECC, result);
 }
 
+static void test_golay2412_message(void)
+{
+    /* source for random bytes: https://www.random.org/bytes */
+    unsigned char data[] = {201, 240, 154, 5, 227, 60, 116, 192, 214};
+    unsigned char result[sizeof(data)];
+    unsigned char msg_enc[2*sizeof(data)];
+
+    golay2412_encode(sizeof(data), &data[0], &msg_enc[0]);
+
+    /* Add errors here. golay(24, 12) can correct up to 3 errors in one 24-bit
+     * symbol (= 3 bytes). Positions for bitflips generated at
+     * https://www.random.org/bytes */
+    msg_enc[0]  ^= (1L << 14) | (1L << 13) | (1L <<  5);
+    msg_enc[3]  ^= (1L <<  0) | (1L << 21) | (1L << 18);
+    msg_enc[6]  ^= (1L << 16) | (1L << 22) | (1L << 19);
+    msg_enc[9]  ^= (1L << 11) | (1L <<  8) | (1L << 15);
+    msg_enc[12] ^= (1L <<  9) | (1L <<  2) | (1L << 10);
+    msg_enc[15] ^= (1L << 17) | (1L <<  1) | (1L <<  7);
```

Comment:
Some interesting "corner-case" maybe would here to also have only 2 errors or 1 error or no errors not always 3 ;-).